lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +26,235 @@ def load_team_membership_csv(course, response):
         "Team membership CSV download is not yet implemented."
     )
     response.write(not_implemented_message + "\n")
+
+
+class TeamMembershipImportManager(object):
+    """
+    A manager class that is responsible the import process of csv file including validation and creation of
+    team_courseteam and teams_courseteammembership objects.
+    """
+
+    def __init__(self, course):
+        self.validation_errors = []
+        self.teamset_ids = []
+        self.user_ids_by_teamset_id = {}
+        self.teamset_ids = []
+        self.number_of_record_added = 0
+        # stores the course for which we are populating teams
+        self.course = course
+        self.max_errors = 0
+        self.existing_course_team_memberships = {}
+        self.existing_course_teams = {}
+
+    @property
+    def import_succeeded(self):
+        """
+        Helper wrapper that tells us the status of the import
+        """
+        return not self.validation_errors
+
+    def set_team_membership_from_csv(self, input_file):
+        """
+        Assigns team membership based on the content of an uploaded CSV file.
+        Returns true if there were no issues.
+        """
+        reader = csv.DictReader((line.decode('utf-8-sig').strip() for line in input_file.readlines()))
+        self.teamset_ids = reader.fieldnames[2:]
+        row_dictionaries = []
+        csv_usernames = set()
+        if not self.validate_teamsets():
+            return False
+        self.load_user_ids_by_teamset_id()
+        self.load_course_team_memberships()
+        self.load_course_teams()
+        # process student rows:
+        for row in reader:
+            username = row['user']
+            if not username:
+                continue
+            if not self.check_for_duplicate_usernames_on_input(username, csv_usernames):
```

Comment:
Nit: also for a function like this where you're returning a bool, I like the `is` naming convention so the conditional reads better.

```suggestion
            if not self.is_username_is_duplicated(username, csv_usernames):
```